docs/tutorials/Containers.md

```diff
@@ -0,0 +1,475 @@
+    #gcr-api-queries-per-100-seconds = 1000
+
+    # Time in minutes before an entry expires from the docker hashes cache and needs to be fetched again
+    #cache-entry-ttl = "20 minutes"
+
+    # Maximum number of elements to be kept in the cache. If the limit is reached, old elements will be removed from the cache
+    #cache-size = 200
+
+    # How should docker hashes be looked up. Possible values are "local" and "remote"
+    # "local": Lookup hashes on the local docker daemon using the cli
+    # "remote": Lookup hashes on docker hub and gcr
+    #method = "remote"
+  }
+}
+```
+
+
+### Best Practices
+
+#### Image Versions
+
+When choosing the image version for your pipeline stages, it is highly recommended that you use a hash rather than a tag, for the sake of reproducibility
+For example, in WDL, you could do this:
+```wdl
+runtime {
+    docker: 'ubuntu:latest'
+}
+```
+
+But what you should do is this:
+```wdl
+runtime {
+    docker: 'ubuntu@sha256:7a47ccc3bbe8a451b500d2b53104868b46d60ee8f5b35a24b41a86077c650210'
+}
+```
+
+You can find the `sha256` of an image using `docker images --digests`
+ 
+ 
+### Notes
+
+#### How does Cromwell know when a job or container has completed?
+Cromwell uses the presence of the `rc` (returncode) file to determine whether a task has succeeded or failed. This `rc` file is generated as part of the `script` within the execution directory, where the script is assembled at runtime. This is important as if the script executes successfully but the container doesn't terminate, Cromwell will continue the execution of the workflow and the container will persist hogging system resources.
+
+Within the two configurations above:
+- `singularity`: The exec mode does not run a container on the background
+- `udocker`: the `--rm` flag is included so that the container is terminated after execution script.
+
+#### Cromwell: Run-in-background
+
+By enabling Cromwell's run-in-background mode, you remove the necessity for the `kill`, `check-alive` and `job-id-regex` blocks, which disables some safety checks when running workflows:
+
+- If there is an error starting the container or executing the script, Cromwell may not recognise this error and hang. For example, this may occur if the container attempts to exceed its allocated resources (runs out of memory); the container daemon may terminate the container without completing the script.
+- If you abort the workflow (by attempting to close Cromwell or issuing an abort command), Cromwell does not have a reference to the job and will not be able to execute the container.
+
+This is only necessary in local environments where there is no job manager to control this, however if your container technology can emit an identifier to stdout, then you are able to remove the run-in-background flag. Note that the check-alive block is not called to check a job-status, rather Cromwell looks for the presence of an `rc` file (see the previous section for more information).
```

Comment:
I'm going to remove:
> Note that the check-alive block is not called to check a job-status, rather Cromwell looks for the presence of an `rc` file (see the previous section for more information). 

This seems to be changed in [cromwell-37](https://github.com/broadinstitute/cromwell/issues/4560), so I don't want to make any assertions.